teos-common/src/lightning/messages.rs

```diff
@@ -0,0 +1,355 @@
+//! Watchtower custom lightning messages that implement LDK's [`Readable`] & [`Writeable`] traits.
+//!
+//! [`Readable`]: lightning::util::ser::Readable
+
+use crate::appointment::Locator;
+use crate::lightning::ser_macros::{impl_writeable_msg, set_msg_type};
+use bitcoin::secp256k1::PublicKey;
+use bitcoin::Txid;
+use lightning::io::Error;
+use lightning::ln::wire;
+use lightning::util::ser::{Writeable, Writer};
+
+// Re-exporting this for other crates to use.
+pub use crate::lightning::ser_utils::Type;
+
+/// The register message sent by the user to subscribe for the watching service.
+#[derive(Debug)]
+pub struct Register {
+    pub pubkey: PublicKey,
+    pub appointment_slots: u32,
+    pub subscription_period: u32,
+}
+
+/// The subscription details message that is sent to the user after registering or toping up.
+/// This message is the response to the register message.
+#[derive(Debug)]
+pub struct SubscriptionDetails {
+    pub appointment_max_size: u16,
+    pub start_block: u32,
+    pub amount_msat: u32,
+    // Optional TLV.
+    pub invoice: Option<String>,
+    pub signature: Option<String>,
+}
+
+/// The add/update appointment message sent by the user.
+#[derive(Debug)]
+pub struct AddUpdateAppointment {
+    pub locator: Locator,
+    // NOTE: LDK will prefix varying size fields (e.g. vectors and strings) with their length.
+    pub encrypted_blob: Vec<u8>,
+    pub signature: String,
+    // Optional TLV.
+    pub to_self_delay: Option<u64>,
+}
+
+/// The appointment accepted message that is sent after an accepted add/update appointment message.
+#[derive(Debug)]
+pub struct AppointmentAccepted {
+    pub locator: Locator,
+    pub start_block: u32,
+    // Optional TLV.
+    pub receipt_signature: Option<String>,
+}
+
+/// The appointment rejected message that is sent if an add/update appointment message was rejected.
+#[derive(Debug)]
+pub struct AppointmentRejected {
+    pub locator: Locator,
+    pub rcode: u16,
+    pub reason: String,
+}
+
+/// The get appointment message sent by the user to retrieve a previously sent appointment from the tower.
+#[derive(Debug)]
+pub struct GetAppointment {
+    pub locator: Locator,
+    pub signature: String,
+}
+
+/// The appointment data message sent by the tower after a get appointment message.
+#[derive(Debug)]
+pub struct AppointmentData {
+    pub locator: Locator,
+    pub encrypted_blob: Vec<u8>,
+}
+
+/// The tracker data message sent by the tower when the requested appointment has been acted upon.
+#[derive(Debug)]
+pub struct TrackerData {
+    pub dispute_txid: Txid,
+    pub penalty_txid: Txid,
+    pub penalty_rawtx: Vec<u8>,
+}
```

Comment:
> BOLT-wise this should be two different messages right?

Indeed they are.

> Have you put any thought on how this would look like?

Do you mean in terms of return types of methods and the typing stuff in Rust?
They are both `TowerMessage` at the end.